generic-tabs/GenericTabs.js

```diff
@@ -70,11 +70,29 @@ export class GenericTabs extends SelectedMixin(BatchingElement) {
 
   constructor() {
     super();
+
+    // Check if slots were setup
+    // If not, assume heading structure and preprocess light DOM
+    if (!this.querySelector('[slot]')) {
```

Comment:
contents aren't available at this stage in parse, so this won't work if someone registered this in the head and it completed before parsed - does it matter?

Reply:
I could try and move it back to the `update()` function (where that comment I accidentally left is).

Reply:
yeah, idk... slotchange is great in some ways for this stuff because it always happens, regardless - so even if people added things early, late, dynamically, etc - but it requires you to kind of 'rebalance' them and prevent infinite recursion as you do.

Reply:
I hit the endless recursion problem when moving this to `slotchange`. But I got dynamic updates working in the `update()` function. It's all local right now because I can't figure out a good test working for it yet.